internal/commands/user/input_test.go

```diff
@@ -96,3 +100,174 @@ func TestProviderTypeDisplayUser(t *testing.T) {
+			Disabled:   true,
+			Data:       map[string]interface{}{"email": "user-3@test.com"},
+		},
+	}
+
+	t.Run("when finding users", func(t *testing.T) {
+		for _, tc := range []struct {
+			description   string
+			inputs        multiUserInputs
+			users         []realm.User
+			expectedUsers []realm.User
+			expectedErr   error
+		}{
+			{
+				description:   "should return found all users with no input users ",
+				users:         testUsers,
+				expectedUsers: testUsers,
+			},
+			{
+				description:   "should error with input users and no found users",
+				inputs:        multiUserInputs{Users: []string{"user-1"}},
+				expectedUsers: nil,
+				expectedErr:   errors.New("no users found"),
+			},
+
+			{
+				description:   "should return an empty users slice with no found users or input users",
+				expectedUsers: nil,
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				realmClient := mock.RealmClient{}
+				realmClient.FindUsersFn = func(groupID, appID string, filter realm.UserFilter) ([]realm.User, error) {
+					return tc.users, nil
+				}
+
+				users, err := tc.inputs.findUsers(realmClient, "groupID", "appID")
+				assert.Equal(t, tc.expectedErr, err)
+				assert.Equal(t, tc.expectedUsers, users)
+			})
+		}
+	})
+
+	t.Run("should error from client", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+		realmClient.FindUsersFn = func(groupID, appID string, filter realm.UserFilter) ([]realm.User, error) {
+			return nil, errors.New("client error")
+		}
+		inputs := multiUserInputs{}
+		_, err := inputs.findUsers(realmClient, "groupID", "appID")
+
+		assert.Equal(t, errors.New("client error"), err)
+	})
+}
+
+func TestMultiUsersInputsSelect(t *testing.T) {
+	testUsers := []realm.User{
+		{
+			ID:         "user-1",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeAnonymous}},
+		},
+		{
+			ID:         "user-2",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeUserPassword}},
+			Disabled:   true,
+			Data:       map[string]interface{}{"email": "user-2@test.com"},
+		},
+		{
+			ID:         "user-3",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeUserPassword}},
+			Disabled:   true,
+			Data:       map[string]interface{}{"email": "user-3@test.com"},
+		},
+	}
+
+	t.Run("when selecting users", func(t *testing.T) {
+		for _, tc := range []struct {
+			description   string
+			inputs        multiUserInputs
+			procedure     func(c *expect.Console)
+			users         []realm.User
+			expectedUsers []realm.User
+		}{
+			{
+				description: "should prompt with no input set",
+				procedure: func(c *expect.Console) {
+					c.ExpectString("Which user(s) would you like to delete?")
+					c.Send("user-1")
+					c.SendLine(" ")
+					c.ExpectEOF()
+				},
+				users:         testUsers,
+				expectedUsers: []realm.User{testUsers[0]},
+			},
+			{
+				description: "should not prompt if no users are found",
```

Comment:
I missed your comment on this before... when have tests that assert the absence of prompts, we don't need to write them in this style where we're using a pseudo-terminal and all.  we probably want to move things around here to be:

```go
t.Run("should prompt with no input set", func(t *testing.T) {
  _, console, _, ui, consoleErr := mock.NewVT10XConsole()
  assert.Nil(t, consoleErr)
  defer console.Close()

  doneCh := make(chan (struct{}))
  go func() {
    defer close(doneCh)
 
    console.ExpectString("Which user(s) would you like to delete?")
    console.Send("user-1")
    console.SendLine(" ")
    console.ExpectEOF()
  }()

  var i multiUserInputs
  users, err := i.selectUsers(ui, testUsers, "delete")

  console.Tty().Close() // flush the writers
  <-doneCh              // wait for procedure to complete

  assert.Nil(t, err)
  assert.Equal(t, testUsers[0:1], users)
})

t.Run("should not prompt the user", func(t *testing.T) {
for _, tc := range []struct{
  description string
  inputs multiUserInputs
  users []realm.User
}{
  {"with no inputs set and no users found"},
  {
    "with user inputs set and some users found",
    multiUserInputs{Users: []string{"user-1"}},
    testUsers,
  },
}{
  t.Run(tc.description, func(t *testing.T) {
    users, err := tc.inputs.selectUsers(nil, tc.users, "")
    assert.Nil(t, err)
    assert.Equal(t, tc.users, users)
  })
}
})
```